packages/cli/src/config/setup/Setup.handler.ts

```diff
@@ -0,0 +1,218 @@
+                    selectedStyle: this.term.brightGreen,
+                    leftPadding: "  ",
+                    selectedLeftPadding: "> ",
+                    submittedLeftPadding: "> ",
+                }) as unknown as {
+                    // biome-ignore lint/suspicious/noExplicitAny: Required for callback
+                    on: (event: string, handler: (response: any) => void) => void;
+                    abort: () => void;
+                    select: (index: number) => void;
+                };
+
+                // Menu selection logic
+                const moveSelection = (direction: 1 | -1) => {
+                    let newIndex = selectedIndex;
+                    do {
+                        newIndex += direction;
+                        if (newIndex < 0) newIndex = opts.items.length - 1;
+                        if (newIndex >= opts.items.length) newIndex = 0;
+                    } while (opts.items[newIndex]?.separator && newIndex >= 0 && newIndex < opts.items.length);
+
+                    selectedIndex = newIndex;
+                    menu.select(selectedIndex);
+                };
+
+                // Key bindings for navigation
+                const keyHandler = (key: string) => {
+                    if (key === "UP" || key === "k") moveSelection(-1);
+                    if (key === "DOWN" || key === "j") moveSelection(1);
+                    if (key === "TAB") moveSelection(1);
+                };
+                this.term.on("key", keyHandler);
+
+                // Handle menu submission
+                // biome-ignore lint/suspicious/noExplicitAny: Required for callback
+                menu.on("submit", (response: any) => {
+                    const selected = opts.items[response.selectedIndex];
+
+                    // Cleanup event listeners and input grabbing
+                    this.term.removeListener("key", keyHandler);
+                    this.term.grabInput(false);
+                    resolve(selected?.separator ? undefined : selected);
+                });
+
+                // Handle cancellation or ESCAPE
+                const cancelHandler = (key: string) => {
+                    if (key === "ESCAPE" || key === "CTRL_C") {
+                        if (key === "CTRL_C") this.term.moveTo(1, y + opts.items.length + 2);
+                        // Cleanup event listeners and input grabbing
+                        this.term.removeListener("key", keyHandler);
+                        this.term.removeListener("key", cancelHandler);
+                        menu.abort();
+                        this.term.grabInput(false);
+                        resolve(undefined);
+                    }
+                };
+                this.term.on("key", cancelHandler);
+            });
+        });
+    }
+
+    protected getCurrentDir(): string | undefined {
+        return process.cwd();
+    }
+
+    protected getProfileType(): IProfileTypeConfiguration[] {
```

Comment:
This name may potentially be confusing since we have methods elsewhere with a similar name (`getProfileType` in ZE API returns a type string like "ssh"). Probably being pick, but would it make sense to rename this function on the abstract class to something like `getProfileSchemas`?